crates/thetawave_interface/src/objective.rs

```diff
@@ -1,9 +1,65 @@
 use bevy_ecs::prelude::Event;
 use serde::Deserialize;
 
+use crate::spawnable::{MobSegmentType, MobType};
+
 // Event for sending damage dealt from mob reaching bottom of arena
 #[derive(Event)]
-pub struct MobReachedBottomGateEvent(pub DefenseInteraction);
+pub struct MobReachedBottomGateEvent {
+    pub mob_type: Option<MobType>,
+    pub mob_segment_type: Option<MobSegmentType>,
+    pub defense_interaction: DefenseInteraction,
+}
```

Comment:
Hmm. Lets talk about this type. Should this have an enum? Or just an `Entity`? 2 options seems odd.